### PR TITLE
Correct background iact image

### PR DESCRIPTION
**Pull request by registerrier**
This PR is meant to provide a correct IACTBasicImageEstimator with better acceptance map calculation as well as a FoVBackgroundEstimator.

Before:
capture d ecran 2018-02-28 a 23 28 57

After:
figure_2

with Fov background:
figure_3

### CODE REVIEW

**Comment by cdeil** [comment]
@registerrier - I wanted to do the v0.7 release now. OK to put this for the next release? It will be in a few weeks, we need stable releases now. We have other bugs e.g. in 1D spec analysis, so from my side getting one or two more PRs in v0.7 is not important.

When you open a PR, can you please always comment if you're posting it for feedback, or if it's ready for review / merging from your side. Obviously tests that show if this works are missing, but given that we want to write new code based on `gammapy.maps` to do this anyways very soon, I'd be OK to just merge this in now as-is without tests.

File: gammapy/image/basic.py [hunk 1]
```diff
@@ -260,26 +263,69 @@ def _counts(self, observation):
...
+            number of bins for the energy integration
+        """
+        p = self.parameters
+        ebins = EnergyBounds.equal_log_spacing(p['emin'], p['emax'], nbins)
+        acceptance = SkyImage.empty_like(counts)
+        acceptance.name = 'exposure_on'
+
+        offsets = observation.pointing_radec.separation(acceptance.coordinates())
+
+        # This is a somewhat dirty approach to deal with the different background IRFs
```
Comment (cdeil) [inline gammapy/image/basic.py]: You could just add a uniform evaluate on the two background classes for now as a new method, with any name and signature you like. We will have to work on IRF classes for weeks / months, I don't think we should be shy to edit them, we will have to iterate there, and anyways IRFs should be considered low-level, not super stable API for now that most users don't use directly.

**Comment by mackaiver** [comment]
Nice. This means we can remove the old/wrong one right?

File: gammapy/image/basic.py (hunk 1 above)
Comment (registerrier) [reply gammapy/image/basic.py]: I suppose we won't be supporting EnergyOffsetArray for long so I am not sure it is worth changing the API for this one, no?

Comment (cdeil) [reply gammapy/image/basic.py]: Agreed!

File: gammapy/background/ring.py [hunk 2]
```diff
@@ -409,3 +410,49 @@ def ring_alpha(theta, r_in, r_out):
...
+
+        if self.norm:
+            counts_excluded = np.sum(counts.data * exclusion.data)
+            acceptance_excluded = np.sum(exposure_on.data * exclusion.data)
+            #print(counts_excluded, acceptance_excluded, counts_excluded / acceptance_excluded)
+            norm = counts_excluded / acceptance_excluded
+        else:
+            norm=1
+        result = SkyImageList()
+        result['background'] = SkyImage(data=norm * exposure_on.data, wcs=wcs)
```
[the remark this answers is not in the document]
Comment (registerrier) [reply gammapy/background/ring.py]: Done

**Comment by registerrier** [comment]
This updated version includes corrections for the review. 
The test was also failing because of the absence of bkg IRF in some GAMMAPY_EXTRA data. A test is added that will assume a flat acceptance map in case no bkg IRF can be found. 
Note that this will create bad results in case a FoVBackgroundEstimator is used without normalization on the counts.

**Pull request merged**